doc/book/app_server/luajit_memprof.rst

```diff
@@ -0,0 +1,524 @@
+
+Frequently Asked Questions
+--------------------------
+
+In this section, some of the profiler-related points are discussed in
+a Q&A format.
+
+**Question (Q)**: Is the profiler suitable for C allocations or allocations
+inside C code?
+
+**Answer (A)**: The profiler reports only allocation events caused by the Lua
+allocator. All Lua-related allocations, like table or string creation
+are reported. But the profiler doesn't report allocations made by ``malloc()``
+or other non-Lua allocators. You can use ``valgrind`` to debug them.
+
+|
+
+**Q**: Why are there so many ``INTERNAL`` allocations in my profiling report?
+What does it mean?
+
+**A**: ``INTERNAL`` means that these allocations/reallocations/deallocations are
+related to the internal LuaJIT structures or are made on JIT traces.
+Currently, the memory profiler doesn't report verbosely allocations of objects
+that are made during trace execution. Try to :ref:`add jit.off() <profiler_usage_internal_jitoff>`
+before profiler start.
+
+|
+
+**Q**: Why is there some reallocations/deallocations without the ``Overrides``
+section?
+
+**A**: These objects can be created before the profiler starts. Adding
+``collectgarbage()`` before the profiler's start enables to collect all
+previously allocated objects that are dead when the profiler starts.
+
+|
+
+**Q**: Why some objects are not collected during profiling? Is it
+a memory leak?
+
+**A**: LuaJIT uses incremental Garbage Collector (GC). A GC cycle may not be
+finished at the moment of the profiler's stop. Add ``collectgarbage()`` before
+stopping the profiler to collect all the dead objects for sure.
+
+|
+
+**Q**: Can I profile not just a current chunk but the entire running application?
+Can I start the profiler when the application is already running?
+
+**A**: Yes. Here is the example of code that can be inserted in the Tarantool
+console for a running instance.
+
+..  code-block:: lua
+    :linenos:
+
+    local fiber = require "fiber"
+    local log = require "log"
+
+    fiber.create(function()
+      fiber.name("memprof")
+
+      collectgarbage() -- Collect all objects already dead
+      log.warn("start of profile")
+
+      local st, err = misc.memprof.start(FILENAME)
+      if not st then
+        log.error("failed to start profiler: %s", err)
+      end
+
+      fiber.sleep(TIME)
+
+      collectgarbage()
+      st, err = misc.memprof.stop()
+
+      if not st then
+        log.error("profiler on stop error: %s", err)
+      end
+
+      log.warn("end of profile")
+    end)
+
+where
+
+*   ``FILENAME``—a name of the report file in binary format
+*   ``TIME``—duration of profiling, seconds.
+
+Also, you can directly call ``misc.memprof.start()`` and ``misc.memprof.stop()``
+from a console.
+
+.. _profiler_analysis:
+
+Report analysis example
+-----------------------
+
+In the example below, the following Lua code named ``format_concat.lua`` is
+investigated with the help of the memory profiler reports.
+
+..  code-block:: lua
```

Comment:
@Buristan, please adjust this chunk considering our style guide.

Reply:
```lua
-- Prevent allocations on new traces.
jit.off()

local function concat(a)
  local nstr = a.."a"
  return nstr
end

local function format(a)
  local nstr = string.format("%sa", a)
  return nstr
end

collectgarbage()

local binfile = "/tmp/memprof_"..(arg[0]):match("([^/]*).lua")..".bin"

local st, err = misc.memprof.start(binfile)
assert(st, err)

-- Payload.
for i = 1, 10000 do
  local f = format(i)
  local c = concat(i)
end
collectgarbage()

local st, err = misc.memprof.stop()
assert(st, err)

os.exit()

```

Also, it is necessary to update all profiler's reports (see below).

Reply:
corrected